src/from-kml/tracepoints-converter.ts

```diff
@@ -63,47 +73,81 @@ import * as fastXmlParser from "fast-xml-parser";
  * ```json
  * [
  *   {
- *     "Position": [12.419255, 41.899689],
- *     "Speed": 36,
- *     "Timestamp": "2013-07-15T10:24:52Z"
+ *     "Position": [-122.419424, 37.774930],
+ *     "Timestamp": "2024-11-19T14:45:00Z"
  *   },
  *   {
- *     "Position": [12.420505, 41.900891],
- *     "Speed": 36,
- *     "Timestamp": "2013-07-15T10:24:52Z"
+ *     "Position": [-122.420157, 37.775032],
+ *     "Timestamp": "2024-11-19T14:45:00Z"
  *   }
  * ]
  * ```
  */
-export function kmlStringToRoadSnapTracePointList(kmlString: string) {
-  const xmlParser = new fastXmlParser.XMLParser({
-    ignoreAttributes: false,
-    attributeNamePrefix: "@_",
-  });
 
-  const result = xmlParser.parse(kmlString);
-  const placemarks = Array.isArray(result.kml.Document.Placemark)
-    ? result.kml.Document.Placemark
-    : [result.kml.Document.Placemark];
+export function kmlStringToRoadSnapTracePointList(content: string): RoadSnapTracePoint[] {
+  const dom = new JSDOM(content, {
+    contentType: "text/xml",
+  });
 
-  return placemarks.flatMap((placemark) => {
-    if (placemark.Point) {
-      const [lon, lat] = placemark.Point.coordinates.split(",").map(Number);
+  // Convert KML to GeoJSON
+  const geoJson = tj.kml(dom.window.document);
+  console.log("First feature properties:", JSON.stringify(geoJson.features[0].properties, null, 2));
+  const features: Feature<Point>[] = geoJson.features.flatMap((feature) => {
+    console.log("Feature properties:", JSON.stringify(feature.properties, null, 2));
+    if (feature.geometry.type === "Point") {
+      const [lon, lat] = (feature.geometry as Point).coordinates;
       return [
         {
-          Position: [parseFloat(lon), parseFloat(lat)],
+          type: "Feature",
+          geometry: {
+            type: "Point",
+            coordinates: [lon, lat],
+          },
+          properties: {
+            ...(feature.properties.timestamp && {
+              timestamp_msec: new Date(feature.properties.timestamp).getTime(),
+            }),
+          },
         },
       ];
-    } else if (placemark.LineString) {
-      const coordinates = placemark.LineString.coordinates.trim().split(/\s+/);
-      return coordinates.map((coord) => {
-        const [lon, lat] = coord.split(",").map(Number);
-        return {
-          Position: [parseFloat(lon), parseFloat(lat)],
-        };
-      });
-    } else {
-      console.log("Invalid input: unrecognized placemark format'");
+    } else if (feature.geometry.type === "LineString") {
+      return feature.geometry.coordinates.map((coord) => ({
+        type: "Feature",
+        geometry: {
+          type: "Point",
+          coordinates: [coord[0], coord[1]], // Take only longitude and latitude
```

Comment:
JFYI (not actually suggesting this):

```suggestion
          coordinates: coord.slice(0, 2), // Take only longitude and latitude
```

Reply:
Thanks!